adapters/Unity/UnityAdapter/GADMUnityInterstitialAd.m

```diff
@@ -118,73 +118,28 @@ - (void)presentInterstitialFromRootViewController:(UIViewController *)rootViewCo
+    NSLog(@"Unity Ads received call to show before successfully loading an ad");
   }
-
-  [UnityAds addDelegate:self];
+  
   [strongConnector adapterWillPresentInterstitial:strongAdapter];
-  [UnityAds show:rootViewController placementId:_placementID];
+  [UnityAds show:rootViewController placementId:_placementID showDelegate:self];
 }
 
-#pragma mark - Unity Delegate Methods
-
-- (void)unityAdsPlacementStateChanged:(NSString *)placementId
-                             oldState:(UnityAdsPlacementState)oldState
-                             newState:(UnityAdsPlacementState)newState {
-}
+#pragma mark - UnityAdsLoadDelegate Methods
 
-- (void)unityAdsDidFinish:(NSString *)placementID withFinishState:(UnityAdsFinishState)state {
-  [UnityAds removeDelegate:self];
+- (void)unityAdsAdLoaded:(nonnull NSString *)placementId {
+  _placementID = placementId;
   id<GADMAdNetworkConnector> strongNetworkConnector = _connector;
   id<GADMAdNetworkAdapter> strongAdapter = _adapter;
-  if (strongNetworkConnector && strongAdapter) {
-    [strongNetworkConnector adapterWillDismissInterstitial:strongAdapter];
-    [strongNetworkConnector adapterDidDismissInterstitial:strongAdapter];
-  }
-}
 
-- (void)unityAdsDidClick:(NSString *)placementID {
-  id<GADMAdNetworkConnector> strongNetworkConnector = _connector;
-  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
-  // The Unity Ads SDK doesn't provide an event for leaving the application, so the adapter assumes
-  // that a click event indicates the user is leaving the application for a browser or deeplink, and
-  // notifies the Google Mobile Ads SDK accordingly.
   if (strongNetworkConnector && strongAdapter) {
-    [strongNetworkConnector adapterDidGetAdClick:strongAdapter];
-    [strongNetworkConnector adapterWillLeaveApplication:strongAdapter];
-  }
-}
-
-- (void)unityAdsDidError:(UnityAdsError)error withMessage:(NSString *)message {
-  [UnityAds removeDelegate:self];
-  id<GADMAdNetworkConnector> strongNetworkConnector = _connector;
-  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
-
-  if (error == kUnityAdsErrorShowError) {
-    if (strongNetworkConnector && strongAdapter) {
-      [strongNetworkConnector adapterWillDismissInterstitial:strongAdapter];
-      [strongNetworkConnector adapterDidDismissInterstitial:strongAdapter];
-    }
+    [strongNetworkConnector adapterDidReceiveInterstitial:strongAdapter];
   }
 }
 
-- (void)unityAdsDidStart:(nonnull NSString *)placementId {
-  // nothing to do
-  // todo: double check if impression need this
-}
-
-- (void)unityAdsReady:(nonnull NSString *)placementId {
-  // Logic to mark a placement ready has moved to the UnityAdsLoadDelegate function
-  // unityAdsAdLoaded.
-}
-
-#pragma mark - UnityAdsLoadDelegate Methods
-
-- (void)unityAdsAdFailedToLoad:(nonnull NSString *)placementId {
+- (void)unityAdsAdFailedToLoad:(NSString *)placementId withError:(UnityAdsLoadError)error withMessage:(NSString *)message {
+  _placementID = placementId;
```

Comment:
per my other comment of renaming the ivar to _loadedPlacementID, you can now set _loadedPlacementID to nil when the ad fails to load, and just use placementId directly through the rest of this method.

Reply:
I think we would prefer to still set the placement ID here so that if show is called despite the load failure, we will gain insight that this happened.  Otherwise show will be called with nil and we won't know what placement ID was attempting to be shown.

Reply:
Ok, perhaps my advice to use "_loadedPlacementID" set from callbacks isn't good then.

Instead of trying to set a _placementID ivar from your callbacks, I suggest you just set the _placementID ivar to ```[[[strongConnector credentials] objectForKey:kGADMAdapterUnityPlacementID] copy];``` for interstitals, and ```_adConfiguration.credentials.settings[kGADMAdapterUnityPlacementID];``` for rewarded at the time when the ad is loaded. That ID comes from the string the user pasted into the AdMob UI for your "placementID" and is already the ID being used to load the ad. That way, you'll never get asked to show an ad with a nil placement, and if the ad isn't loaded at show time, your SDK is probably already logging something useful in the show failure callback.